functions/xcluster/main.k

```diff
@@ -0,0 +1,327 @@
+
+import models.io.upbound.platform.aws.v1alpha1 as awsplatformv1alpha1
+import models.io.upbound.platform.observe.v1alpha1 as observev1alpha1
+import models.io.upbound.platform.gitops.v1alpha1 as gitopsv1alpha1
+
+oxr = option("params").oxr
+_ocds = option("params").ocds
+_dxr = option("params").dxr
+dcds = option("params").dcds
+
+_metadata = lambda name: str -> any {
+    { annotations = { "krm.kcl.dev/composition-resource-name" = name }}
+}
+
+# Extract parameters from the observed XR
+_id = oxr.spec?.parameters?.id
```

Comment:
Would be nice if we could omit the leading `_` if variable is not mutable. Minor nit but improves clarity imho